backend/src/main/java/com/redhat/cloud/notifications/db/repositories/DrawerNotificationRepository.java

```diff
@@ -0,0 +1,138 @@
+            .setParameter("readStatus", readStatus)
+            .setParameter("notificationIds", notificationIds)
+            .executeUpdate();
+    }
+
+    public List<DrawerEntryPayload> getNotifications(String orgId, String username, Set<UUID> bundleIds, Set<UUID> appIds, Set<UUID> eventTypeIds,
+                                              LocalDateTime startDate, LocalDateTime endDate, Boolean readStatus, Query query) {
+        query.setSortFields(DrawerNotification.SORT_FIELDS);
+        query.setDefaultSortBy("created:DESC");
+        Optional<Query.Sort> sort = query.getSort();
+
+        String hql = "SELECT dn.id, dn.read, " +
+            "dn.event.bundleDisplayName, dn.event.applicationDisplayName, dn.event.eventTypeDisplayName, dn.created, dn.event.renderedDrawerNotification "
+            + "FROM DrawerNotification dn where dn.orgId = :orgId and dn.userId = :userid";
+
+        hql = addHqlConditions(hql, bundleIds, appIds, eventTypeIds, startDate, endDate, readStatus);
+        if (sort.isPresent()) {
+            hql += getOrderBy(sort.get());
+        }
+
+        TypedQuery<Object[]> typedQuery = entityManager.createQuery(hql, Object[].class);
+        setQueryParams(typedQuery, orgId, username, bundleIds, appIds, eventTypeIds, startDate, endDate, readStatus);
+
+        Query.Limit limit = query.getLimit();
+
+        typedQuery.setMaxResults(limit.getLimit());
+        typedQuery.setFirstResult(limit.getOffset());
+
+        List<Object[]> results = typedQuery.getResultList();
+        return  results.stream().map(e -> new DrawerEntryPayload(e)).collect(Collectors.toList());
+    }
+
+    public Long count(String orgId, String username, Set<UUID> bundleIds, Set<UUID> appIds, Set<UUID> eventTypeIds,
+                      LocalDateTime startDate, LocalDateTime endDate, Boolean readStatus) {
+
+        String hql = "SELECT count(dn.id) "
+            + "FROM DrawerNotification dn where dn.orgId = :orgId and dn.userId = :userid";
+
+        hql = addHqlConditions(hql, bundleIds, appIds, eventTypeIds, startDate, endDate, readStatus);
+
+        TypedQuery<Long> typedQuery = entityManager.createQuery(hql, Long.class);
+        setQueryParams(typedQuery, orgId, username, bundleIds, appIds, eventTypeIds, startDate, endDate, readStatus);
+
+        return typedQuery.getSingleResult();
+    }
+
+    private static String addHqlConditions(String hql, Set<UUID> bundleIds, Set<UUID> appIds, Set<UUID> eventTypeIds,
+                                           LocalDateTime startDate, LocalDateTime endDate, Boolean readStatus) {
+
+        if (eventTypeIds != null && !eventTypeIds.isEmpty()) {
+            hql += " AND dn.event.eventType.id IN (:eventTypeIds)";
+        } else if (appIds != null && !appIds.isEmpty()) {
+            hql += " AND dn.event.applicationId IN (:appIds)";
+        } else if (bundleIds != null && !bundleIds.isEmpty()) {
+            hql += " AND dn.event.bundleId IN (:bundleIds)";
+        }
```

Comment:
In the Event Log API, if `bundleIds` and `appIds` contain values, we use an `OR` operator between them. I don't know exactly why but I guess this was requested to implement something in the frontend.

We might want a similar behavior here.

Reply:
We discussed this in a meeting. The current code is fine for V1.